Source/WebCore/editing/TextIterator.cpp

```diff
@@ -455,7 +464,10 @@ static bool isRendererAccessible(const RenderObject* renderer, TextIteratorBehav
     if (renderer->isSkippedContent()) {
         if (!behaviors.contains(TextIteratorBehavior::EntersSkippedContentRelevantToUser))
             return false;
-        return style.usedContentVisibility() == ContentVisibility::Auto;
+        if (style.usedContentVisibility() == ContentVisibility::Auto)
+            return true;
+        if (renderer->document().settings().detailsAutoExpandEnabled() && renderer->style().autoRevealsWhenFound())
+            return true;
```

Comment:
Why does `autoRevealsWhenFound` need to be precomputed and stored on the RenderStyle? Why not just check         `m_document->settings().detailsAutoExpandEnabled() && is<HTMLDetailsElement>(element)` here, through an appropriate helper function? Similarly, why do the `detailsAutoExpandEnabled()` check twice? Is this solving a performance problem?

Reply:
This is a performance optimization, since the `autoRevealsWhenFound` bit is inherited, this avoids a whole ancestor walk:

```
            for (auto& ancestor : ancestorsOfType<RenderElement>(*renderer)) {
                if (RefPtr element = dynamicDowncast<HTMLElement>(ancestor.element()); is<HTMLDetailsElement>(element))
                    return true;
                }
            }
````